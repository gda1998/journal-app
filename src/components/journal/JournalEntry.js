import dayjs from 'dayjs';
import { useDispatch } from 'react-redux';
import { activeNoteAction } from '../../actions/notes';


export const JournalEntry = ({ id, title, body, date, url }) => {
    const noteDate = dayjs(date);
    const dispatch = useDispatch();
    const handleEntryClick = () => dispatch( activeNoteAction(id, { title, body, date, url }) );

    return (
        <div 
            className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
            onClick={ handleEntryClick }
        >

            { url && 
                <div 
                    className="journal__entry-picture"
                    style={{ 
                        backgroundSize: 'cover',
                        backgroundImage: `url(${ url })`
                    }}
                >
                </div>/* /.journal__entry-picture */
            }


            <div className="journal__entry-body">
                <p className="journal__entry-title">
                    { title }
                </p>
                <p className="journal__entry-content">
                    { body }
                </p>
            </div>{/* /.journal__entry-body */}

            <div className="journal__entry-date-box">
                <span>{ noteDate.format('dddd') }</span>
                <h4>{ noteDate.format('DD') }</h4>
            </div>{/* /.journal__entry-date-box */}

        </div>/* /.journal__entry */
    );
}