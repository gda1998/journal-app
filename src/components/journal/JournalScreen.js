import { useSelector } from 'react-redux';
import { Navigate } from 'react-router-dom';
import { NoteScreen } from '../notes/NoteScreen';
import { Sidebar } from './Sidebar';
import { NothingSelected } from './NothingSelected';

export const JournalScreen = () => {
    const { auth, notes } = useSelector( state => state );
    return Object.keys(auth).length === 0 
    ? <Navigate to="/auth/login" />
    : (
        <div className="journal__main-content animate__animated animate__fadeIn animate__faster">
            <Sidebar />
            <main>
                {
                    notes.active === null ? <NothingSelected /> : <NoteScreen />
                }
            </main>
        </div>
    );
}
