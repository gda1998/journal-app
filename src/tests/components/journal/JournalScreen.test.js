import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { JournalScreen } from '../../../components/journal/JournalScreen';

const mockStore = configureStore([thunk]);
const store = mockStore({
    auth: { uid: '1243', name: 'Gabriel' },
    ui: { loading: false, msgError: null },
    notes: {
        active: {},
        notes: [{ id:'11111', date: 1647071655611 }, { id: '22222',date: 1647071655611 }]
    }
});

describe('Pruebas en <JournalScreen />', () => { 
    let wrapper;
    const getWrapper = (store) => mount(
        <Provider store={store}>
            <MemoryRouter initialEntries={['/']}>
                <Routes>
                    <Route path="/" element={ <JournalScreen /> } />
                    <Route path="/auth/login" element={ <h1>LoginScreen</h1> } />
                </Routes>
            </MemoryRouter>
        </Provider>
    );

    beforeEach( () => {
        wrapper = getWrapper(store);
    });

    test('Se debe de mostrar correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de retornar al login si no esta autenticado', () => {
        const store = mockStore({
            auth: {},
            ui: { loading: false, msgError: null }
        });
        wrapper = getWrapper(store);
        expect(wrapper.find('h1').exists()).toBe(true);
        expect(wrapper.find('h1').text()).toBe('LoginScreen');
    });

    test('Debe de mostrarse el Sidebar si esta autenticado', () => { 
        expect(wrapper.find('Sidebar').exists()).toBe(true);
    });
});