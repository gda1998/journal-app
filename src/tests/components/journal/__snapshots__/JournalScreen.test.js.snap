// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalScreen /> Se debe de mostrar correctamente 1`] = `
<div
  className="journal__main-content animate__animated animate__fadeIn animate__faster"
>
  <aside
    className="journal__sidebar"
  >
    <div
      className="journal__sidebar-navbar"
    >
      <h3
        className="mt-5"
      >
        <i
          className="far fa-moon"
        />
        <span>
          Gabriel
        </span>
      </h3>
      <button
        className="btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
    <div
      className="journal__new-entry"
      onClick={[Function]}
    >
      <i
        className="far fa-calendar-plus fa-5x"
      />
      <p
        className="mt-5"
      >
        New Entry
      </p>
    </div>
    <div
      className="journal__entries"
    >
      <div
        className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
        onClick={[Function]}
      >
        <div
          className="journal__entry-body"
        >
          <p
            className="journal__entry-title"
          />
          <p
            className="journal__entry-content"
          />
        </div>
        <div
          className="journal__entry-date-box"
        >
          <span>
            Saturday
          </span>
          <h4>
            12
          </h4>
        </div>
      </div>
      <div
        className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
        onClick={[Function]}
      >
        <div
          className="journal__entry-body"
        >
          <p
            className="journal__entry-title"
          />
          <p
            className="journal__entry-content"
          />
        </div>
        <div
          className="journal__entry-date-box"
        >
          <span>
            Saturday
          </span>
          <h4>
            12
          </h4>
        </div>
      </div>
    </div>
  </aside>
  <main>
    <div
      className="notes__main-content"
    >
      <div
        className="notes__appbar"
      >
        <span>
          05 de Noviembre de 2021
        </span>
        <input
          id="fileSelector"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
        <div>
          <button
            className="btn"
            onClick={[Function]}
          >
            Picture
          </button>
          <button
            className="btn"
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
      <div
        className="notes__content"
      >
        <input
          autoComplete="off"
          className="notes__title-input"
          name="title"
          onChange={[Function]}
          placeholder="Some awesome title"
          type="text"
        />
        <textarea
          className="notes__textarea"
          name="body"
          onChange={[Function]}
          placeholder="What's happened today?"
          type="text"
        />
      </div>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </main>
</div>
`;
